service/history/replication/executable_task_converter.go

```diff
@@ -70,15 +73,19 @@ func (e *executableTaskConverterImpl) Convert(
 			metrics.ToClusterIDTag(clientShardKey.ClusterID),
 			metrics.OperationTag(TaskOperationTag(replicationTask)),
 		)
-		tasks[index] = e.convertOne(taskClusterName, replicationTask)
+		task, err := e.convertOne(taskClusterName, replicationTask)
+		if err != nil {
```

Comment:
@wxing1292 This will basically ignore the task. Is it okay to do so? The [code](https://github.com/temporalio/temporal/blob/main/service/history/replication/executable_history_task.go#L180-L199) here seems we are ignoring the task that failed to deserialize, but want to double confirm with you.

Reply:
you should do the task deserialization within the task, so we can mark the task as poison pill

the code mentioned in the comment only try to get some info for logging